meta/statement_executor.go

```diff
@@ -26,7 +26,7 @@ type StatementExecutor struct {
 		CreateUser(name, password string, admin bool) (*UserInfo, error)
 		UpdateUser(name, password string) error
 		DropUser(name string) error
-		SetPrivilege(username, database string, p influxql.Privilege) error
+		SetPrivilege(username, database string, p influxql.Privilege, admin bool) error
```

Comment:
I'll open a new PR to see if extending `influxql.Privilege` is a better way to pass along the admin privilege (I suspect it is).

Adding the admin flag seemed cleaner when I initially started this PR, but it means the only way to tell if the admin privilege should be set is by checking if the database is empty.